docs/Contributing/architecture/system_design/database.md

```diff
@@ -0,0 +1,78 @@
+[Back to top](./README.md)
+# Database
+
+## MySQL
+
+Primary system of record. Schema evolved via Go migration files in `server/datastore/mysql/migrations/*` (tables + data). Access via datastore layer (`server/datastore/mysql/*.go`) implementing interface methods in `server/fleet` (e.g., `ListHosts`, `NewMDMAppleDeclaration`). SQL composed using `sqlx`, named queries, and small hand-written statements for performance-critical paths.
+
+### Setup / configuration
+Dev: docker-compose spins MySQL 8 with defaults. Test suites set `MYSQL_TEST=1` for integration tests; migrations auto-run on `New` datastore creation. Prod: configure pool sizes (`mysql.max_open_conns`, `max_idle_conns`, `conn_max_lifetime`), optional read replica (`replica` section) used for heavy read endpoints (software listings, vulnerability queries) – selected via `reader` context in datastore.
+
+### Connection lifetimes
+`config.MysqlConfig` holds DSN components; TLS optional (custom CA, certs). Long-running cron tasks keep short transactions; expensive bulk inserts (async host processing) use batches sized by config.
+
+### Host scale expectations
+| Hosts | Pattern | Notes |
+|-------|---------|-------|
+| <5k | synchronous host detail writes | OK to keep immediate updates |
+| 5k–50k | enable async host processing | Redis queue reduces write amplification |
```

Comment:
Is that something we officially recommend and see much use? I thought it was an obscure feature that wasn't used much, but I may be wrong.